wrangler-core/src/main/java/io/cdap/wrangler/statistics/BasicStatistics.java

```diff
@@ -55,10 +58,15 @@ public Row aggregate(List<Row> rows) {
           if (value.isEmpty()) {
             stats.increment(column, "empty");
           } else {
+            // this call is very expensive for string > 2000 characters, took seconds to return
             Map<String, List<String>> finds = engine.findWithType(value);
             for (String find : finds.keySet()) {
               types.increment(column, find);
             }
+            // TODO: this is a workaround for CDAP-18262, to proper fix we should revisit this computation logic
+            if (System.currentTimeMillis() - now > TIME_OUT_MILLIS) {
```

Comment:
now is a confusing variable name, better to name it start or something like that

Reply:
Fixed